### PR TITLE
Family id argument

### PR DESCRIPTION
**Pull request by jose-oc**

### CODE REVIEW

**Comment by jose-oc** [comment]
I forgot to squash but it's just a checkbox on the github pull request page and github will do automatically for us 😉

File: src/client.py [hunk 1]
```diff
@@ -45,25 +46,28 @@ def _login(email: str, password: str):
...
-    def get_classes(self, target_day: datetime):
+    def get_classes(self, target_day: datetime, family_id: str = None):
         response = self.session.get(
             classes_endpoint(self.box_name),
             params={
                 "box": self.box_id,
                 "day": target_day.strftime("%Y%m%d"),
-                "familyId": "",
+                "familyId": family_id,
+                "_": time.time(),
```
Comment (pablobuenaposada) [inline src/client.py:56]: what is this for? is strictly necessary?

File: src/client.py [hunk 2]
```diff
@@ -45,25 +46,28 @@ def _login(email: str, password: str):
...
 
-    def get_classes(self, target_day: datetime):
+    def get_classes(self, target_day: datetime, family_id: str = None):
         response = self.session.get(
             classes_endpoint(self.box_name),
             params={
                 "box": self.box_id,
                 "day": target_day.strftime("%Y%m%d"),
-                "familyId": "",
+                "familyId": family_id,
```
Comment (pablobuenaposada) [inline src/client.py:55]: did you try in production what happens if we send familyId: null? (sorry I don't have access to aimharder anymore)

File: src/client.py [hunk 3]
```diff
@@ -45,25 +46,28 @@ def _login(email: str, password: str):
                 raise IncorrectCredentials
         return session
 
-    def get_classes(self, target_day: datetime):
+    def get_classes(self, target_day: datetime, family_id: str = None):
```
Comment (pablobuenaposada) [inline src/client.py:49]: family_id: str | None = None

File: src/client.py [hunk 4]
```diff
@@ -45,25 +46,28 @@ def _login(email: str, password: str):
...
                 "day": target_day.strftime("%Y%m%d"),
-                "familyId": "",
+                "familyId": family_id,
+                "_": time.time(),
             },
         )
         return response.json().get("bookings")
 
-    def book_class(self, target_day: datetime, class_id: str) -> bool:
+    def book_class(
```
Comment (pablobuenaposada) [inline src/client.py:61]: family_id: str | None = None

File: src/main.py [hunk 5]
```diff
@@ -14,27 +19,36 @@ def get_booking_goal_time(day: datetime, booking_goals):
             booking_goals[str(day.weekday())]["time"],
             booking_goals[str(day.weekday())]["name"],
         )
-    except KeyError:  # did not found a matching booking goal
-        raise NoBookingGoal
+    except KeyError:  # did not find a matching booking goal
+        raise NoClassOnTargetDayTime(
```
Comment (pablobuenaposada) [inline src/main.py:23]: what is the difference between NoBookingGoal and NoClassOnTargetDayTime?

**Review by pablobuenaposada — CHANGES_REQUESTED** [review]

**Comment by pablobuenaposada** [comment]
> I forgot to squash but it's just a checkbox on the github pull request page and github will do automatically for us 😉

File: src/client.py (hunk 1 above)
Comment (jose-oc) [reply src/client.py:56]: I don't think it is strictly necessary but it's how the actual app make the request so I think we should make it like it.

File: src/client.py (hunk 2 above)
Comment (jose-oc) [reply src/client.py:55]: It works on every scenario: when your account only has one member (familyId is empty then), when your account has several members and familyId is empty you use the main member and when you send the familyId you work with that member.

File: src/client.py (hunk 1 above)
Comment (pablobuenaposada) [reply src/client.py:56]: if there's no difference in the response I don't see why having it

**Pull request merged**